Model/ContactClientModel.php

```diff
@@ -306,22 +306,28 @@ public function getStats(
         $dateFormat = null,
         $canViewOthers = true
     ) {
-        $unit           = (null === $unit) ? $this->getTimeUnitFromDateRange($dateFrom, $dateTo) : $unit;
-        $dateToAdjusted = clone $dateTo;
-        if (in_array($unit, ['H', 'i', 's'])) {
-            // draw the chart with the correct intervals for intra-day
-            $dateToAdjusted->setTime(23, 59, 59);
+        if ($dateFrom->getTimezone()->getName() !== 'UTC') {
+            $dateFrom->setTimezone(new \DateTimeZone('UTC'));
         }
-        $chart = new LineChart($unit, $dateFrom, $dateToAdjusted, $dateFormat);
-        $query = new ChartQuery($this->em->getConnection(), $dateFrom, $dateToAdjusted, $unit);
-        $stat  = new Stat();
+        $dateFrom->modify('midnight');
+
+        if ($dateTo->getTimezone()->getName() !== 'UTC') {
+            $dateTo->setTimezone(new \DateTimeZone('UTC'));
+        }
+        $dateTo->modify('tomorrow midnight -1 sec');
```

Comment:
You shouldn't have to do this... getTimeUnitFromDateRange adjusts the datetime provided by reference.
That's what `$dateToAdjusted = clone $dateTo;` was for.